zipline/src/nativeMain/kotlin/app/cash/zipline/nativeUtils.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline
+
+import kotlinx.cinterop.CArrayPointer
+import kotlinx.cinterop.CPointer
+import kotlinx.cinterop.CValues
+import kotlinx.cinterop.CVariable
+import kotlinx.cinterop.NativePlacement
+import kotlinx.cinterop.allocArray
+import kotlinx.cinterop.interpretCPointer
+import kotlinx.cinterop.sizeOf
+
+/** Copy the data of [item] to the [index] of [this] as if it were an array of [T] structs. */
+internal inline operator fun <reified T : CVariable> CPointer<T>.set(index: Int, item: CValues<T>) {
```

Comment:
Note that these types are wider than necessary. Normally you'll have a `CArrayPointer<T>` as the receiver (which is a typealias to `CPointer<T>`) and the item will be a `CValue<T>` (which inherits from `CValues<T>`). I can narrow them if desired, but nothing strictly prevents it since we're down in pointer math and memcpy here.